### ISSUE TITLE
CVE-2022-45146 for java 17

### ISSUE THREAD

**Issue by jayclover** [adoptium/adoptium#256]
### Summary
We are using Prisma scan for below 2 method and it detected one medium finding CVE-2022-45146.
1. Official latest docker image from dockerhub: "**eclipse-temurin:17**"
2. Debian base image + eclipse temurin OpenJDK 17 like below Dockerfile:

```
FROM debian:bullseye

ENV JAVA_HOME=/opt/java/openjdk
COPY --from=eclipse-temurin:17 $JAVA_HOME $JAVA_HOME
ENV PATH="${JAVA_HOME}/bin:${PATH}"
```

### Details
It detected one medium vulnerability: https://nvd.nist.gov/vuln/detail/CVE-2022-45146

### PoC
This vulnerability should come from the openjdk 17.

### Impact
More info, please refer to https://nvd.nist.gov/vuln/detail/CVE-2022-45146

May we know if there is any plan to fix it in new openjdk 17 version? How is the timeline?

**Comment by jayclover** [comment]
**Below is what Prisma detected:**

```
"id": "sha256:6eb30df4a475c19ffe8f08d3bbfe827e89d8d68f9a7deba0609bb7462ffc3f3c",
			"name": "eclipse-temurin:17",
			"distro": "Ubuntu 22.04.3 LTS",
			"distroRelease": "jammy",
			"digest": "sha256:b0bb840379ec3b0f76105ff8c730bd53f346d394ac3c8b09735bfff505c12b1e",
"vulnerabilities": [
				{
					"id": "CVE-2022-45146",
					"status": "fixed in 1.0.2.4",
					"cvss": 5.5,
					"vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:N/A:N",
					"description": "An issue was discovered in the FIPS Java API of Bouncy Castle BC-FJA before 1.0.2.4. Changes to the JVM garbage collector in Java 13 and later trigger an issue in the BC-FJA FIPS modules where it is possible for temporary keys used by the module to be zeroed out while still in use by the module, resulting in errors or potential information loss. NOTE: FIPS compliant users are unaffected because the FIPS certification is only for Java 7, 8, and 11.",
					"severity": "medium",
					"packageName": "java",
					"packageVersion": "17.0.10",
					"link": "https://nvd.nist.gov/vuln/detail/CVE-2022-45146",
					"riskFactors": [
						"Exploit exists - POC",
						"Has fix",
						"Medium severity",
						"Attack complexity: low"
					],
					"impactedVersions": [
						"\u003e=13.0.0"
					],
					"publishedDate": "2022-11-21T12:30:17Z",
					"discoveredDate": "2024-02-19T02:51:22Z",
					"fixDate": "2023-12-15T11:55:32Z",
					"layerTime": "2024-02-13T10:06:28Z",
					"packagePath": "/opt/java/openjdk/bin/java",
					"layerInstruction": "ADD file:7f9a3c5a4231ed19174c21d17ce05d84d568cff6d3a0c2a1d7c3a9be5e45c02c in / "
				}
			],
			"vulnerabilityDistribution": {
				"critical": 0,
				"high": 0,
				"medium": 1,
				"low": 0,
				"total": 1
			},
```

**Comment by karianna** [comment]
This is a CVE for the Bouncy Castle crypto lib, not Eclispe Temurin binaries or Java itself.

**Comment by jayclover** [comment]
> This is a CVE for the Bouncy Castle crypto lib, not Eclispe Temurin binaries or Java itself.

Sorry. May I know how we can avoid this CVE by using eclipse-temurin:17 directly?

**Comment by tellison** [comment]
> Sorry. May I know how we can avoid this CVE by using eclipse-temurin:17 directly?

Eclipse Temurin does not use Bouncy Castle FIPS Java module (BC-FJA) by default, so this CVE will not affect users unless they modify Temurin's security configuration. Furthermore, BC-FJA 1.0.2.3 and earlier are certified for use with Java 7, Java 8, and in the case of the 1.0.2 series Java 11. You have reported this on Java 17, and this issue does not apply where the modules are used on the JVMs that they were tested and certified on.

Therefore you can avoid this CVE by:
 - just using Temurin without configuring it to use the Bouncy Castle FIPS Java module, or
 - only using Temurin with the BC-FJA together with a version of Temurin Java where it is certified, or
 - upgrade the [BC-FJA](https://www.bouncycastle.org/fips_java_roadmap.html) to a later version (note the CVE was fixed in version 1.0.2.4).
